lib/common/output_log.c

```diff
@@ -0,0 +1,267 @@
+log_output_xml(pcmk__output_t *out, const char *name, const char *buf) {
+    xmlNodePtr node = NULL;
+    private_data_t *priv = out->priv;
+    CRM_ASSERT(priv != NULL);
+
+    node = create_xml_node(NULL, name);
+    xmlNodeSetContent(node, (pcmkXmlStr) buf);
+    do_crm_log_xml(LOG_INFO, name, node);
+    free(node);
+}
+
+G_GNUC_PRINTF(4, 5)
+static void
+log_begin_list(pcmk__output_t *out, const char *singular_noun, const char *plural_noun,
+               const char *format, ...) {
+    int len = 0;
+    va_list ap;
+    char* buffer = NULL;
+    private_data_t *priv = out->priv;
+    CRM_ASSERT(priv != NULL);
+
+    va_start(ap, format);
+    len = vasprintf(&buffer, format, ap);
+    CRM_ASSERT(len >= 0);
+    va_end(ap);
+
+    /* Don't skip empty prefixes,
+     * otherwise there will be mismatch 
+     * in the log_end_list */
+    if(strcmp(buffer, "") == 0) {
+        /* nothing */
+    }
+    
+    g_queue_push_tail(priv->prefixes, buffer);
+}
+
+G_GNUC_PRINTF(3, 4)
+static void
+log_list_item(pcmk__output_t *out, const char *name, const char *format, ...) {
+    int len = 0;
+    va_list ap;
+    private_data_t *priv = out->priv;
+    char prefix[LINE_MAX] = { 0 };
+    int offset = 0;
+    char* buffer = NULL;
+
+    CRM_ASSERT(priv != NULL);
+
+    for (GList* gIter = priv->prefixes->head; gIter; gIter = gIter->next) {
+        if (strcmp(prefix, "") != 0) {
+            offset += snprintf(prefix + offset, LINE_MAX - offset, ": %s", (char *)gIter->data);
+        } else {
+            offset = snprintf(prefix, LINE_MAX, "%s", (char *)gIter->data);
+        }
+    }
+
+    va_start(ap, format);
+    len = vasprintf(&buffer, format, ap);
+    CRM_ASSERT(len >= 0);
+    va_end(ap);
+
+    if (strcmp(buffer, "") != 0) { /* We don't want empty messages */
+        if ((name != NULL) && (strcmp(name, "") != 0)) {
+            if (strcmp(prefix, "") != 0) {
+                pcmk__output_crm_log(out, "%s: %s: %s", prefix, name, buffer);
+            } else {
+                pcmk__output_crm_log(out, "%s: %s", name, buffer);
+            }
+        } else {
+            if (strcmp(prefix, "") != 0) {
+                pcmk__output_crm_log(out, "%s: %s", prefix, buffer);
+            } else {
+                pcmk__output_crm_log(out, "%s", buffer);
+            }
+        }
+    }
+    free(buffer);
+}
+
+static void
+log_end_list(pcmk__output_t *out) {
+    private_data_t *priv = out->priv;
+    CRM_ASSERT(priv != NULL);
+    if (priv->prefixes == NULL) {
+      return;
+    }
+    CRM_ASSERT(priv->prefixes->tail != NULL);
+
+    free((char *)priv->prefixes->tail->data);
+    g_queue_pop_tail(priv->prefixes);
+}
+
+G_GNUC_PRINTF(2, 3)
+static void
+log_info(pcmk__output_t *out, const char *format, ...) {
+    int len = 0;
+    va_list ap;
+    char* buffer = NULL;
+
+    va_start(ap, format);
+    len = vasprintf(&buffer, format, ap);
+    CRM_ASSERT(len >= 0);
+    va_end(ap);
+
+    log_list_item(out, NULL, "%s", buffer);
```

Comment:
Why use a list item? Is there a reason to use list prefixes in this case?

Reply:
Fixed. No, there is no actually.